src/ray/object_manager/plasma/plasma_allocator.cc

```diff
@@ -64,13 +66,18 @@ void *PlasmaAllocator::DiskMemalignUnlimited(size_t alignment, size_t bytes) {
   if (!mem) {
     return nullptr;
   }
+  RAY_CHECK(IsOutsideInitialAllocation(mem));
   allocated_ += bytes;
+  fallback_allocated_ += bytes;
   return mem;
 }
 
 void PlasmaAllocator::Free(void *mem, size_t bytes) {
   dlfree(mem);
   allocated_ -= bytes;
+  if (RayConfig::instance().plasma_unlimited() && IsOutsideInitialAllocation(mem)) {
```

Comment:
Why don't we call `IsOutsideInitialAllocation` before dlfree? (in case the function modifies the pointer)? If you know this doesn't it's fine for me

Reply:
The pointer is passed by value, it's impossible for it to be modified.